src/server/db_slice.h

```diff
@@ -388,11 +388,12 @@ class DbSlice {
   // Track keys for the client represented by the the weak reference to its connection.
   void TrackKeys(const facade::Connection::WeakRef&, const ArgSlice&);
 
-  // Delete a key referred by its iterator.
-  void PerformDeletion(PrimeIterator del_it, EngineShard* shard, DbTable* table);
-
  private:
-  void PreUpdate(DbIndex db_ind, PrimeIterator it);
+  enum class PreUpdateMode {
+    kRegular,
+    kNoIncVersion,
+  };
+  void PreUpdate(DbTable* table, PrimeIterator it, PreUpdateMode mode = PreUpdateMode::kRegular);
```

Comment:
I changed to pass in `DbTable` instead of `db_ind` because on flush operations we use a different table (i.e. one that was `move()`d out of `db_arr_`)